src/librbd/Types.h

```diff
@@ -61,8 +61,11 @@ typedef std::map<uint64_t, uint64_t> SnapSeqs;
  * to its parent. This allows copy-on-write images. */
 struct ParentSpec {
   int64_t pool_id;
+  std::string image_name;
```

Comment:
Nit: despite the slight code duplication, I think it would be cleaner to just add the non-parent related bits in a separate ```ImageCtx``` structure

Reply:
@dillaman The only problem to have it as a separate structure looks like is `RefreshParentRequest`. I would need to pass `MigrationSpec` there. Do you want an additional "migration_spec" argument to `RefreshParentRequest`?

Alternatively I could add `MigrationSpec` to `ParentInfo`.

Reply:
It would just be needed for the v1 image name, correct? I guess I'm fine passing it a ```MigrationSpec``` to that class.

Reply:
Yes, it is needed only if the parent is v1 image, to be able to open it by name. Thanks.